src/wr_control_drive_science/src/TeleopScienceControl.cpp

```diff
@@ -0,0 +1,154 @@
+    std::array<float, 3> buf{0, 0, 0};
+    /* Flush anything already in the serial buffer */
+    tcflush(fd, TCIFLUSH);
+    /* read up to 128 bytes from the fd */
+    int readStatus = read(fd, buf.data(), 3*sizeof(float));
+    //NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay, cppcoreguidelines-pro-type-vararg, hicpp-no-array-decay, hicpp-no-assembler, hicpp-vararg)
+    ROS_ASSERT_MSG(readStatus != -1, "Read failed");
+    std_msgs::Float32 temp;
+    temp.data = buf[0];
+    moistureSensor1.publish(temp);
+    temp.data = buf[1];
+    moistureSensor1.publish(temp);
+    temp.data = buf[2];
+    moistureSensor1.publish(temp);
+}
+
+auto main(int argc, char** argv) -> int {
+    ros::init(argc, argv, "Science Teleop Control");
+
+    ros::NodeHandle n;
+    ros::NodeHandle nh{"~"};
+    nh.getParam("fileLoc", file);
+    nh.getParam("baudRate", baudRate);
+    //NOLINTNEXTLINE(cppcoreguidelines-pro-type-vararg, hicpp-vararg, hicpp-signed-bitwise)
+    fd = open(file.c_str(), O_RDWR | O_NOCTTY);
+    //NOLINTNEXTLINE(cppcoreguidelines-pro-bounds-array-to-pointer-decay, cppcoreguidelines-pro-type-vararg, hicpp-no-array-decay, hicpp-no-assembler, hicpp-vararg)
+    ROS_ASSERT_MSG(fd != -1, "File %s does not exist", file.c_str());
+    usleep(3500000);
+    /* Set up the control structure */
+    struct termios toptions;
+ 
+    /* Get currently set options for the tty */
+    tcgetattr(fd, &toptions);
+    
+    /* Set custom options */
+    
+    cfsetispeed(&toptions, baudRate);
+    cfsetospeed(&toptions, baudRate);
+    /* 8 bits, no parity, no stop bits */
+    toptions.c_cflag &= ~PARENB; //NOLINT(hicpp-signed-bitwise)
+    toptions.c_cflag &= ~CSTOPB; //NOLINT(hicpp-signed-bitwise)
+    toptions.c_cflag &= ~CSIZE; //NOLINT(hicpp-signed-bitwise)
+    toptions.c_cflag |= CS8; //NOLINT(hicpp-signed-bitwise)
+
+    /* no hardware flow control */
+    toptions.c_cflag &= ~CRTSCTS;
+
+    /* enable receiver, ignore status lines */
+    toptions.c_cflag |= CREAD | CLOCAL; //NOLINT(hicpp-signed-bitwise)
+
+    /* disable input/output flow control, disable restart chars */
+    toptions.c_iflag &= ~(IXON | IXOFF | IXANY); //NOLINT(hicpp-signed-bitwise)
+
+    /* disable canonical input, disable echo,
+    disable visually erase chars,
+    disable terminal-generated signals */
+    toptions.c_lflag &= ~(ICANON | ECHO | ECHOE | ISIG); //NOLINT(hicpp-signed-bitwise)
+
+    /* disable output processing */
+    toptions.c_oflag &= ~OPOST; //NOLINT(hicpp-signed-bitwise)
+    
+    /* wait for 12 characters to come in before read returns */
+    /* WARNING! THIS CAUSES THE read() TO BLOCK UNTIL ALL */
+    /* CHARACTERS HAVE COME IN! */
+    toptions.c_cc[VMIN] = 12; 
+    /* no minimum time to wait before read returns */
+    toptions.c_cc[VTIME] = 0;
+    
+    /* commit the options */
+    tcsetattr(fd, TCSANOW, &toptions);
+    
+    /* Wait for the Arduino to reset */
+    usleep(1000*1000);
```

Comment:
There may be a way to do this dynamically.

Reply:
I found that the reason this happens is due to a trigger called DTR on the serial line.  See if there is a way to disable it, since we don't really need to reset the Arduino on program start.